Update to CI and env consistency, Export fix for ics

lucs7 opened the pull request:
- environment configuration consistency
- updated composer to include php-cs-fixer
- added parallel-lint for php
- added composer caching for slight speedup
- updated workflow
- moved tox to ci
- moved version to composer.json and added getversion function
- fixed export links for atom and ics

**the move of tox and the python requirements will require an update to the readthedocs script**

**CI workflow improvements and refactoring:**

* Major restructuring of the `.github/workflows/lint-and-analyse-php.yml` workflow:
  - Jobs are renamed for clarity (e.g., `lint` → `basic-checks`, `phpunit` → `lint-php-files`, `markdownlint` → `lint-markdown`). [[1]](diffhunk://#diff-8726c263478e2e8f5194944f4e84bfe220cfd1dd68422100adfe16415a754f0dR15-R21) [[2]](diffhunk://#diff-8726c263478e2e8f5194944f4e84bfe220cfd1dd68422100adfe16415a754f0dL48-R51)
  - Steps now use explicit tox configuration (`tox -c ci/tox.ini --workdir .`) for consistency. [[1]](diffhunk://#diff-8726c263478e2e8f5194944f4e84bfe220cfd1dd68422100adfe16415a754f0dL31-R38) [[2]](diffhunk://#diff-9cf2000c53760d837a449f874e53f792819108d3a4bf346336d0f7d082deae2cL34-R34)
  - Adds a `configKeys-check` step to validate config keys.
  - Improves Composer caching and dependency installation for faster, more reliable builds. [[1]](diffhunk://#diff-8726c263478e2e8f5194944f4e84bfe220cfd1dd68422100adfe16415a754f0dR64-R77) [[2]](diffhunk://#diff-8726c263478e2e8f5194944f4e84bfe220cfd1dd68422100adfe16415a754f0dR120-R198)
  - Linting now uses `parallel-lint` piped to `cs2pr` for better integration with GitHub annotations.
  - Test and analysis jobs are reorganized for better dependency order and parallelism. [[1]](diffhunk://#diff-8726c263478e2e8f5194944f4e84bfe220cfd1dd68422100adfe16415a754f0dR107) [[2]](diffhunk://#diff-8726c263478e2e8f5194944f4e84bfe220cfd1dd68422100adfe16415a754f0dR120-R198)
  - The old `ci/ci-phplint` script is removed in favor of the new linting approach.

**Environment variable and configuration consistency:**

* Updates in `.env.example` for naming consistency and clarity:
  - All plugin-related variables are renamed from `LB_PLUGIN_*` to `LB_PLUGINS_*`.
  - Image upload variables are renamed for clarity (`LB_UPLOAD_IMAGE_DIRECTORY` → `LB_UPLOADS_IMAGE_UPLOAD_DIRECTORY`, etc.).
  - Password reset variable is renamed for consistency (`LB_DISABLE_PASSWORD_RESET` → `LB_PASSWORD_DISABLE_RESET`).
  - Adds new variables for resource options and security headers. [[1]](diffhunk://#diff-a3046da0d15a27e89f2afe639b25748a7ad4d9290af3e7b1b6c1a5533c8f0a8cR377-R383) [[2]](diffhunk://#diff-a3046da0d15a27e89f2afe639b25748a7ad4d9290af3e7b1b6c1a5533c8f0a8cR501-R503)

**Configuration validation scripts:**

* Adds new Python scripts for configuration validation:
  - `ci/configKeys-check.py` checks that all config keys in `config.dist.php` are defined in `ConfigKeys.php`.
  - `ci/config-check.py` is refactored to check that all config keys have corresponding environment variables, improving its robustness and accuracy.
  - `ci/tox.ini` (renamed from `tox.ini`) is updated to include the new `configKeys-check` environment.

**Version management improvements:**

* Updates code to use `Configuration::GetAppVersion()` instead of the `VERSION` constant for retrieving the application version, ensuring a single source of truth and future flexibility. [[1]](diffhunk://#diff-b3cc769564eab3628cb0eef0b8786ad67d6f47be31f2c8e2ec5ee30b6dd63177L62-R62) [[2]](diffhunk://#diff-54e4a1076c86f2b38e467bb9e52046a7f88f6eadace089afe3e72de5a3358d67L113-R113) [[3]](diffhunk://#diff-54e4a1076c86f2b38e467bb9e52046a7f88f6eadace089afe3e72de5a3358d67L126-R128)

---

**CI/CD and Linting Improvements**
- Refactored GitHub Actions workflows for PHP linting, analysis, and testing: clearer job names, improved caching, new config validation steps, and replaced custom lint script with `parallel-lint` and `cs2pr` integration. [[1]](diffhunk://#diff-8726c263478e2e8f5194944f4e84bfe220cfd1dd68422100adfe16415a754f0dR15-R21) [[2]](diffhunk://#diff-8726c263478e2e8f5194944f4e84bfe220cfd1dd68422100adfe16415a754f0dL31-R38) [[3]](diffhunk://#diff-8726c263478e2e8f5194944f4e84bfe220cfd1dd68422100adfe16415a754f0dL48-R51) [[4]](diffhunk://#diff-8726c263478e2e8f5194944f4e84bfe220cfd1dd68422100adfe16415a754f0dR64-R77) [[5]](diffhunk://#diff-8726c263478e2e8f5194944f4e84bfe220cfd1dd68422100adfe16415a754f0dL135-R98) [[6]](diffhunk://#diff-8726c263478e2e8f5194944f4e84bfe220cfd1dd68422100adfe16415a754f0dR107) [[7]](diffhunk://#diff-8726c263478e2e8f5194944f4e84bfe220cfd1dd68422100adfe16415a754f0dR120-R198) [[8]](diffhunk://#diff-8651876ce43320990c467198b0070d88d08750474a3c72a0fa0d89519b1db715L1-L16) [[9]](diffhunk://#diff-9cf2000c53760d837a449f874e53f792819108d3a4bf346336d0f7d082deae2cL34-R34)

**Environment Variable and Config Consistency**
- Standardized environment variable names in `.env.example`, especially for plugins, image uploads, and password reset options; added new variables for resource and security options. [[1]](diffhunk://#diff-a3046da0d15a27e89f2afe639b25748a7ad4d9290af3e7b1b6c1a5533c8f0a8cL164-R167) [[2]](diffhunk://#diff-a3046da0d15a27e89f2afe639b25748a7ad4d9290af3e7b1b6c1a5533c8f0a8cR377-R383) [[3]](diffhunk://#diff-a3046da0d15a27e89f2afe639b25748a7ad4d9290af3e7b1b6c1a5533c8f0a8cL424-R431) [[4]](diffhunk://#diff-a3046da0d15a27e89f2afe639b25748a7ad4d9290af3e7b1b6c1a5533c8f0a8cR501-R503) [[5]](diffhunk://#diff-a3046da0d15a27e89f2afe639b25748a7ad4d9290af3e7b1b6c1a5533c8f0a8cL632-R663)

**Configuration Validation**
- Introduced `ci/configKeys-check.py` to ensure all config keys are present in `ConfigKeys.php`, and refactored `ci/config-check.py` to robustly verify environment variable coverage for config keys. [[1]](diffhunk://#diff-655e9ecce3611db2ff7db33d6ac75fde925c919bfe8ca0e7a89954b98729869eR1-R86) [[2]](diffhunk://#diff-d8aac25c03610ce8ae03a8f89d40ce735dc190ea22173752999994ef1c82da60L7-R89)
- Updated `ci/tox.ini` to include new config validation environments and improved doc build path handling. [[1]](diffhunk://#diff-0b1acd7a6b12d19fd9ad690f5e6ba46e9955f523eadb821cd985cbbea3101cc3L5-R5) [[2]](diffhunk://#diff-0b1acd7a6b12d19fd9ad690f5e6ba46e9955f523eadb821cd985cbbea3101cc3L16-R17)

**Version Handling**
- Replaced direct usage of the `VERSION` constant with `Configuration::GetAppVersion()` throughout the codebase for improved maintainability. [[1]](diffhunk://#diff-b3cc769564eab3628cb0eef0b8786ad67d6f47be31f2c8e2ec5ee30b6dd63177L62-R62) [[2]](diffhunk://#diff-54e4a1076c86f2b38e467bb9e52046a7f88f6eadace089afe3e72de5a3358d67L113-R113) [[3]](diffhunk://#diff-54e4a1076c86f2b38e467bb9e52046a7f88f6eadace089afe3e72de5a3358d67L126-R128)

lucs7 committed:
fix: added php-cs-fixer and phpstan memory limit in composer.json

lucs7 committed:
fix: update environment variable keys and add resource options in config files

lucs7 committed:
fix: updated config-check.py to .env validation

lucs7 committed:
ci: add configKeys-check to validate keys in config files

lucs7 committed:
ci: enhance ci-phplint script with progress indication and error reporting

lucs7 committed:
ci: rename markdownlint job to lint-markdown and reorganize phpunit job dependencies

lucs7 committed:
ci: add workflow_dispatch trigger to lint-and-analyse-php workflow for testing

lucs7 committed:
ci: moved tox to ci

ci: moved tox to ci

ci: moved tox to ci

JohnVillalovos commented on .env.example:
```diff
@@ -374,6 +374,13 @@ LB_REGISTRATION_AUTO_SUBSCRIBE_EMAIL=false
 LB_REGISTRATION_NOTIFY_ADMIN=false
 
+##########################################
+# Ressource Options
```
spelling error.  s/Ressource/Resource/

JohnVillalovos commented on ci/config-check.py:
```diff
@@ -4,63 +4,98 @@
 import dataclasses
 import pathlib
-import pprint
+from typing import Set
```
This isn't needed.

JohnVillalovos commented on ci/config-check.py:
```diff
@@ -4,63 +4,98 @@
 import sys
 
 
+def parse_php_config_array(content: str) -> Set[str]:
```
Should be able to do just:  `-> set[str]`   Not required to use `Set` since Python 3.9 or greater.

lucs7 committed:
ci: add cs2pr and PHP-CS-Fixer

lucs7 committed:
fix: small PR comments and issues

lucs7 committed:
ci: replace custom PHP lint script with php-parallel-lint and update linting command

lucs7 committed:
ci: add caching for Composer packages and update install options

lucs7 committed:
fix: show option key rather than values on confi wrong choice

lucs7 committed:
fix: null error on unknown key

lucs7 committed:
fix: wrong section for slack token

lucs7 committed:
fix: exporter page broken after config validation
exporter pages like atom or ics were dependend on old require.login key that did not exist for a long time. as the call was failing no SecurePageDecorator was applied. As now invalid config keys do not return false but an exaption, the SecurePageDecorator is not applied and the route fails.
Repairs on SecurePageDecorator logic and maybe basic auth follow later

lucs7 committed:
fix: error in reservation.start.time.constraint
'Any' key ways suggested instead of 'none' which does not work
